libsolidity/ast/ASTJsonImporter.h

```diff
@@ -0,0 +1,142 @@
+/*
+	This file is part of solidity.
+
+	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+/**
+ * @author julius <djudju@protonmail.com>
+ * @date 2019
+ * Converts the AST from JSON format to ASTNode
+ */
+
+#pragma once
+
+#include <vector>
+#include <libsolidity/ast/AST.h>
+#include <json/json.h>
+#include <libsolidity/ast/ASTAnnotations.h>
+#include <liblangutil/Exceptions.h>
+#include <liblangutil/SourceLocation.h>
+
+namespace dev
+{
+namespace solidity
+{
+
+/**
+ * Converter of the AST from JSON format to ASTNode
+ */
+class ASTJsonImporter
+{
+public:
+	using SourceLocation = langutil::SourceLocation;
+
+	/// Create an importer to import a given abstract syntax tree in Json format to an ASTNode
+	/// @a _sourceList used to provide source names for the ASTs
+	ASTJsonImporter(std::map<std::string, Json::Value const*> _sourceList);
+
+	/// converts the AST from JSON-format to ASTPointer
+	std::map<std::string, ASTPointer<SourceUnit>> jsonToSourceUnit();
```

Comment:
maybe just `import()`?

Reply:
Also it might be useful to have a private constructor and the only public function being a static function that takes the string-json map and returns a string-sourceUnit map.

Reply:
> just import()?

hmm to me it does not reflect that there is an ouput... (i like the current name, especially if we move the passing of the argument from constructor to the function...

Reply:
removed public constructor and moved the argument to jsonToSourceUnit()